bloom_filter.py

```diff
@@ -23,6 +25,11 @@ def hash_function(self):
         pass
 
     
-    def calculate_no_hash_functions(self):
+    def calculate_no_hash_functions(self, size_of_bit, length_of_element):
         """Calculates the optimal number of hash functions."""
-        pass
\ No newline at end of file
+        length_of_element = len(self.input_item)
+        size_of_bit = int(-(length_of_element * math.log(self.probability_false_positive))/(math.log(2)**2))
+        return size_of_bit
+        no_of_hash_functions = int((size_of_bit/length_of_element)* math.log(2))
+        return no_of_hash_functions
+        
```

Comment:
Maybe make two functions out of this? one calculating the optimal size and the other calculating the optimal number of hash functions?